diff --git a/examples/example-server-lib/decoration_provider.cpp b/examples/example-server-lib/decoration_provider.cpp
index 4d72a978d7..34ca1a3b35 100644
--- a/examples/example-server-lib/decoration_provider.cpp
+++ b/examples/example-server-lib/decoration_provider.cpp
@@ -45,7 +45,8 @@ void null_window_callback(MirWindow*, void*) {}
 
 struct preferred_codecvt : std::codecvt_byname<wchar_t, char, std::mbstate_t>
 {
-    preferred_codecvt() : std::codecvt_byname<wchar_t, char, std::mbstate_t>("") {}
+    preferred_codecvt() :
+        std::codecvt_byname<wchar_t, char, std::mbstate_t>(std::locale("").name().c_str()) {}
     ~preferred_codecvt() = default;
 };
 
