diff --git a/src/Dash/CMakeLists.txt b/src/Dash/CMakeLists.txt
index cb00caf25..952ae1114 100644
--- a/src/Dash/CMakeLists.txt
+++ b/src/Dash/CMakeLists.txt
@@ -16,6 +16,7 @@ set_target_properties(unity8-dash PROPERTIES INSTALL_RPATH "${CMAKE_INSTALL_PREF
 target_link_libraries(unity8-dash
     Qt5::DBus Qt5::Gui Qt5::Qml Qt5::Quick Qt5::Test
     UbuntuGestures connectivity-qt1
+    intl
 )
 
 if (ENABLE_TOUCH_EMULATION)
diff --git a/tools/scopetool/CMakeLists.txt b/tools/scopetool/CMakeLists.txt
index 751d021f6..50fdf33d8 100644
--- a/tools/scopetool/CMakeLists.txt
+++ b/tools/scopetool/CMakeLists.txt
@@ -4,7 +4,7 @@ add_executable(${SCOPE_TOOL}
     ../../src/UnixSignalHandler.cpp
     )
 
-target_link_libraries(${SCOPE_TOOL} Qt5::Qml Qt5::Quick)
+target_link_libraries(${SCOPE_TOOL} Qt5::Qml Qt5::Quick intl)
 
 # install binaries
 install(TARGETS ${SCOPE_TOOL}
diff --git a/plugins/LightDM/FullLightDM/CMakeLists.txt b/plugins/LightDM/FullLightDM/CMakeLists.txt
index 0d0154172..e9c804bff 100644
--- a/plugins/LightDM/FullLightDM/CMakeLists.txt
+++ b/plugins/LightDM/FullLightDM/CMakeLists.txt
@@ -27,6 +27,7 @@ target_link_libraries(FullLightDM-qml
     ${LIBLIGHTDM_LDFLAGS}
     ${LIBUSERMETRICSOUTPUT_LDFLAGS}
     Qt5::DBus Qt5::Gui Qt5::Qml
+    intl
     )
 
 add_unity8_plugin(FullLightDM 0.1 LightDM/FullLightDM TARGETS FullLightDM-qml)
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index eae303268..5d4c38c0b 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -52,6 +52,7 @@ target_link_libraries(${SHELL_APP}
     Qt5::DBus Qt5::Gui Qt5::Qml Qt5::Quick Qt5::Test
     ${ANDROID_PROPERTIES_LDFLAGS} ${GSETTINGS_QT_LDFLAGS}
     UbuntuGestures connectivity-qt1 unity8-private
+    intl
     )
 
 if (ENABLE_TOUCH_EMULATION)
diff --git a/tests/plugins/LightDM/IntegratedLightDM/CMakeLists.txt b/tests/plugins/LightDM/IntegratedLightDM/CMakeLists.txt
index c35cbff8f..ccc0d5617 100644
--- a/tests/plugins/LightDM/IntegratedLightDM/CMakeLists.txt
+++ b/tests/plugins/LightDM/IntegratedLightDM/CMakeLists.txt
@@ -15,6 +15,7 @@ add_dependencies(GreeterDBusTestExec MockLightDM)
 target_link_libraries(GreeterDBusTestExec
     Qt5::Core Qt5::DBus Qt5::Quick Qt5::Test
     MockLightDM
+    intl
     )
 target_include_directories(GreeterDBusTestExec PUBLIC
     ${CMAKE_SOURCE_DIR}/plugins/LightDM
diff --git a/tests/plugins/LightDM/IntegratedLightDM/CMakeLists.txt b/tests/plugins/LightDM/IntegratedLightDM/CMakeLists.txt
index ccc0d5617..8d8d07a07 100644
--- a/tests/plugins/LightDM/IntegratedLightDM/CMakeLists.txt
+++ b/tests/plugins/LightDM/IntegratedLightDM/CMakeLists.txt
@@ -88,6 +88,7 @@ add_dependencies(GreeterUsersModelTestExec MockLightDM)
 target_link_libraries(GreeterUsersModelTestExec
     Qt5::Core Qt5::Test
     MockLightDM
+    intl
     )
 target_include_directories(GreeterUsersModelTestExec PUBLIC
     ${CMAKE_SOURCE_DIR}/plugins/LightDM
diff --git a/tests/plugins/LightDM/IntegratedLightDM/CMakeLists.txt b/tests/plugins/LightDM/IntegratedLightDM/CMakeLists.txt
index 8d8d07a07..67918cfc8 100644
--- a/tests/plugins/LightDM/IntegratedLightDM/CMakeLists.txt
+++ b/tests/plugins/LightDM/IntegratedLightDM/CMakeLists.txt
@@ -116,6 +116,7 @@ add_dependencies(GreeterPromptsModelTestExec MockLightDM)
 target_link_libraries(GreeterPromptsModelTestExec
     Qt5::Core Qt5::Test
     MockLightDM
+    intl
     )
 target_include_directories(GreeterPromptsModelTestExec PUBLIC
     ${CMAKE_SOURCE_DIR}/plugins/LightDM
diff --git a/tests/mocks/AccountsService/CMakeLists.txt b/tests/mocks/AccountsService/CMakeLists.txt
index c10e2dab8..b6a3a4563 100644
--- a/tests/mocks/AccountsService/CMakeLists.txt
+++ b/tests/mocks/AccountsService/CMakeLists.txt
@@ -17,6 +17,7 @@ add_library(MockAccountsService-qml MODULE
 target_link_libraries(MockAccountsService-qml
     Qt5::DBus Qt5::Qml
     MockLightDM
+    intl
     )
 
 add_unity8_mock(AccountsService 0.1 AccountsService
